### PR TITLE
Enable server when not building client

### PR DESCRIPTION
**Pull request by SmallJoker**
``` HTML
<Zeno`>-BDUILD_CLIENT=false should automatically add DBUILD_SERVER=true
<Zeno`>easy to add but I am scared to modify CMakeLists.txt
<Zeno`>people get angry about it too easily lol
<Zeno`>Krock, you add it, I'll commit
```

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -30,11 +30,20 @@ mark_as_advanced(EXECUTABLE_OUTPUT_PATH LIBRARY_OUTPUT_PATH)
...
+endif()
+
 if(ENABLE_CURL)
 	find_package(CURL)
 	if (CURL_FOUND)
 		message(STATUS "cURL support enabled.")
 		set(USE_CURL TRUE)
+	else(BUILD_SERVER)
+		message(WARNING "cURL is required to announce to the server list")
+	else(BUILD_CLIENT)
```
Comment (PilzAdam) [inline src/CMakeLists.txt]: `BUILD_CLIENT` and `BUILD_SERVER` are not exclusive. If both are set, both warning messages should be displayed.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -30,11 +30,20 @@ mark_as_advanced(EXECUTABLE_OUTPUT_PATH LIBRARY_OUTPUT_PATH)
...
+	set(BUILD_SERVER TRUE)
+	set(ENABLE_CURL TRUE)
+endif()
+
 if(ENABLE_CURL)
 	find_package(CURL)
 	if (CURL_FOUND)
 		message(STATUS "cURL support enabled.")
 		set(USE_CURL TRUE)
+	elseif(BUILD_SERVER)
```
Comment (est31) [inline src/CMakeLists.txt]: I don't think this does what @PilzAdam has requested.

**Comment by Calinou** [comment]
Is it possible to make such a warning appear in yellow, to be more noticeable?

**Comment by est31** [comment]
If, then colors should be added to all warnings, and errors.

File: src/CMakeLists.txt (hunk 2 above)
Comment (est31) [inline src/CMakeLists.txt]: Now it does, thanks.

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -30,11 +30,23 @@ mark_as_advanced(EXECUTABLE_OUTPUT_PATH LIBRARY_OUTPUT_PATH)
 option(ENABLE_CURL "Enable cURL support for fetching media" TRUE)
 set(USE_CURL FALSE)
 
+if(NOT BUILD_CLIENT)
+	set(BUILD_SERVER TRUE)
+	set(ENABLE_CURL TRUE)
```
Comment (est31) [inline src/CMakeLists.txt]: Why set this to true? its already defaulting to true, isn't it? If people want to turn it off, let them.

**Comment by est31** [comment]
@Zeno- don't you want to :+1: it?

File: src/CMakeLists.txt (hunk 3 above)
Comment (rubenwardy) [inline src/CMakeLists.txt]: Cached settings?

If you do this:

    cmake . -DBUILD_SERVER=0
    cmake . -DBUILD_CLIENT=0   # -DBUILD_SERVER=0 is cached here.

Afaik.

Comment (est31) [inline src/CMakeLists.txt]: The comment is about `ENABLE_CURL`.

**Comment by kilbith** [comment]
Needs a rebase and label `[Needs second review]`

File: src/CMakeLists.txt [hunk 4]
```diff
@@ -30,6 +30,10 @@ mark_as_advanced(EXECUTABLE_OUTPUT_PATH LIBRARY_OUTPUT_PATH)
 option(ENABLE_CURL "Enable cURL support for fetching media" TRUE)
 set(USE_CURL FALSE)
 
+if(NOT BUILD_CLIENT)
+	set(BUILD_SERVER TRUE)
```
Comment (ShadowNinja) [inline src/CMakeLists.txt]: IMO there should be a warning here:
```CMake
if(NOT (BUILD_CLIENT OR BUILD_SERVER))
	message(WARNING "Neither BUILD_CLIENT nor BUILD_SERVER set!  Setting BUILD_SERVER.")
	set(BUILD_SERVER TRUE)
```

**Comment by SmallJoker** [comment]
Rebased and added warning, based on SN's suggestion

**Comment by nerzhul** [comment]
@SmallJoker okay for me

**Comment by PilzAdam** [comment]
#3120

**Comment by ShadowNinja** [comment]
7419504079c894dd15fd84116505b853feb44287, e511282d78a844af24602791cd3cab9aa6e6996d

**Pull request closed without merging**